Dockerfile

```diff
@@ -0,0 +1,35 @@
+# This Dockerfile builds an image containing the Mac and Windows version of kube-compare
+# layered on top of the Linux cli image.
+FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.22-builder-multi-openshift-4.17 AS builder-rhel-8
+WORKDIR /go/src/github.com/openshift/kube-compare
+COPY . .
+RUN go work vendor
+RUN make cross-build --warn-undefined-variables
+
+FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.22-builder-multi-openshift-4.17 AS builder-rhel-9
+WORKDIR /go/src/github.com/openshift/kube-compare
+COPY . .
+RUN go work vendor
+RUN make cross-build --warn-undefined-variables
+
+FROM --platform=linux/amd64 registry.ci.openshift.org/ocp/4.17:cli
+
+COPY --from=builder-rhel-9 /go/src/github.com/openshift/kube-compare/_output/bin/ /usr/share/openshift/
+
+RUN cd /usr/share/openshift && \
+    ln -sf /usr/share/openshift/linux_amd64/kube-compare /usr/bin/kube-compare && \
+    mv windows_amd64 windows && \
+    mv darwin_amd64 mac && \
+    mv darwin_arm64 mac_arm64
+
+COPY --from=builder-rhel-8 /go/src/github.com/openshift/kube-compare/_output/bin/linux_amd64/kubectl-cluster_compare /usr/share/openshift/linux_amd64/kube-compare.rhel8
+COPY --from=builder-rhel-8 /go/src/github.com/openshift/kube-compare/_output/bin/linux_arm64/kubectl-cluster_compare /usr/share/openshift/linux_arm64/kube-compare.rhel8
+COPY --from=builder-rhel-8 /go/src/github.com/openshift/kube-compare/_output/bin/linux_ppc64le/kubectl-cluster_compare /usr/share/openshift/linux_ppc64le/kube-compare.rhel8
+COPY --from=builder-rhel-8 /go/src/github.com/openshift/kube-compare/_output/bin/linux_s390x/kubectl-cluster_compare /usr/share/openshift/linux_s390x/kube-compare.rhel8
+
+COPY --from=builder-rhel-9 /go/src/github.com/openshift/kube-compare/_output/bin/linux_amd64/kubectl-cluster_compare /usr/share/openshift/linux_amd64/kube-compare.rhel9
+COPY --from=builder-rhel-9 /go/src/github.com/openshift/kube-compare/_output/bin/linux_arm64/kubectl-cluster_compare /usr/share/openshift/linux_arm64/kube-compare.rhel9
+COPY --from=builder-rhel-9 /go/src/github.com/openshift/kube-compare/_output/bin/linux_ppc64le/kubectl-cluster_compare /usr/share/openshift/linux_ppc64le/kube-compare.rhel9
+COPY --from=builder-rhel-9 /go/src/github.com/openshift/kube-compare/_output/bin/linux_s390x/kubectl-cluster_compare /usr/share/openshift/linux_s390x/kube-compare.rhel9
```

Comment:
As a final step...can we have the base image as scratch? https://hub.docker.com/_/scratch/ 

This help with any security concerns and perfect for our use case since this simply a storage for us.

Reply:
In our case, the binary depends on the arch. How can we use scratch for this?

Reply:
As I understand you should use scratch when you want to build a base image, not sure if that's our case.
Also, if I put it in the end of the dockerfile it looks like you can't access the shell.

Reply:
gotcha!